File: 2887-sort-vowels-in-a-string/sort-vowels-in-a-string.js
var sortVowels = function(s) {
    var vowelSet = new Set(['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U']);
    var vowList = [];

    for (var i = 0; i < s.length; i++) {
        var c = s[i];
        if (vowelSet.has(c)) {
            vowList.push(c);
        }
    }

    vowList.sort();

    var res = [];
    var v = 0;
    for (var i = 0; i < s.length; i++) {
        var char = "";
        var c = s[i];
        if (!vowelSet.has(c)) {
            char = c;
        } else {
            char = vowList[v];
            v++;
        }
        res.push(char);
    }

    return res.join("");
};

File: 2887-sort-vowels-in-a-string/sort-vowels-in-a-string.java
class Solution {
    public String sortVowels(String s) {
        HashSet<Character> vowSet = new HashSet<>(Arrays.asList(new Character[]{'a','e','i','o','u','A','E','I','O','U'}));
        List<Character> vowList = new ArrayList<>();
        for(Character c : s.toCharArray()){
            if(vowSet.contains(c))
                vowList.add(c);
        }

        vowList.sort(null);
        // System.out.println(vowList);
        StringBuilder res = new StringBuilder();
        int  p= 0; 
        for(Character c: s.toCharArray()){
            if (!vowSet.contains(c))
                res.append(c);
            else{
                Character vow = vowList.get(p);
                p+=1;
                res.append(vow);

            }
           
        }


        return res.toString();


    }
}